public/app/core/components/TransformersUI/setGeometry/setGeometry.test.ts

```diff
@@ -0,0 +1,40 @@
+import { FieldMatcherID, fieldMatchers, FieldType } from '@grafana/data';
+import { toDataFrame } from '@grafana/data/src/dataframe/processDataFrame';
+import { DataTransformerID } from '@grafana/data/src/transformations/transformers/ids';
+import { frameAsGazetter } from 'app/features/geo/gazetteer/gazetteer';
+
+describe('Set geometry', () => {
+  it('adds lat/lon based on string field', async () => {
+    const cfg = {
+      id: DataTransformerID.setGeometry,
+      options: {
+        lookupField: 'location',
+        gazetteer: 'public/gazetteer/usa-states.json',
+      },
+    };
+    const data = toDataFrame({
+      name: 'locations',
+      fields: [
+        { name: 'location', type: FieldType.string, values: ['AL', 'AK', 'Arizona', 'Arkansas', 'Somewhere'] },
+        { name: 'values', type: FieldType.number, values: [0, 10, 5, 1, 5] },
+      ],
+    });
+
+    const matcher = fieldMatchers.get(FieldMatcherID.byName).get(cfg.options?.lookupField);
+
+    const frame = toDataFrame({
+      fields: [
+        { name: 'id', values: ['AL', 'AK', 'AZ'] },
+        { name: 'name', values: ['Alabama', 'Arkansas', 'Arizona'] },
+        { name: 'lng', values: [-80.891064, -100.891064, -111.891064] },
+        { name: 'lat', values: [12.448457, 24.448457, 33.448457] },
+      ],
+    });
+    const gaz = frameAsGazetter(frame, { path: 'path/to/gaz.json' });
+
+    // TODO!!!!!
```

Comment:
Is this to do comment needed still? Might make sense to fully complete this test before merging in PR

Reply:
I'll pull the `setGeometry` transformer into the next PR